CMakeLists.txt

```diff
@@ -143,6 +143,16 @@ else()
   find_package(GLEW REQUIRED)
 endif()
 
+IF(NOT HAVE_ASIO OR HAVE_ASIO STREQUAL "")
+  unset(HAVE_ASIO CACHE)
+  include(CheckIncludeFileCXX)
+  CHECK_INCLUDE_FILE_CXX("asio.hpp" HAVE_ASIO)
+  IF(NOT HAVE_ASIO)
+    message(FATAL_ERROR "Please install the standaline Asio library from https://think-async.com/Asio/")
```

Comment:
I think we should add the instructions here too, or to the wiki and link it here.